lib/upload/index.coffee

```diff
@@ -52,30 +56,63 @@ exports.exec = ({design, user, password, host}={}, done) ->
     assert(typeof password is 'string', "The parameter 'password' is required")
     assert(typeof host is 'string', "The parameter 'host' is required")
   catch err
-    return done(err)
+    return callback(err)
 
-  authenticate {host, user, password}, (err, res) ->
+  exports.authenticate {host, user, password}, (err, {user, accessToken:token}={}) ->
+    return callback(err) if err
+    exports.putJson {design, token, host}, (err, design) ->
+      return callback(err) if err
+      exports.uploadAssets {cwd, design, host, token}, (err) ->
+        return callback(err) if err
+        callback(null, {design})
+
+
+exports.putJson = ({design, host, token}, callback) ->
+  request
+    method: 'put'
+    url: host+"/designs/#{design.name}/#{design.version}"
+    headers: Authorization: "Bearer #{token}"
+    body: design
+    json: true
+  , (err, res, body) ->
+    return callback(null, body) if res?.statusCode == 200
     if err
-      return done(err)
+      console.error(err)
+      return callback(err)
+    else
+      error = new Error(body.error || "Unhandled response code #{statusCode}")
+      error.error_details = body.error_details
+      callback(error)
+
 
-    css = []
-    for asset in design?.assets?.css || []
-      css.push(url.resolve("http://livingdocs-designs.s3.amazonaws.com/#{design.name}/#{design.version}/", asset))
-    design.assets.css = css if css.length
+exports.uploadAssets = ({cwd, design, host, token}, callback) ->
+  Glob = require('glob')
+  new Glob '**/*', cwd: cwd, (err, files) ->
+    return callback(err) if err
 
-    request
-      method: 'put'
-      url: host+"/designs/#{design.name}/#{design.version}"
-      headers: Authorization: "Bearer #{res.access_token}"
-      body: design
-      json: true
-    , (err, res, body) ->
-      return done() if res?.statusCode == 200
-      if err
-        console.error(err)
-        return done(err)
-      else
-        console.log(body)
-        error = new Error(body.error)
-        error.error_details = body.error_details
-        done(error)
+    files = files.filter (file) -> return !/^design.js(on)?$/.test(file)
+    async.eachLimit files, 10, (file, done) ->
+      file = path.join(cwd, file)
+      fs.stat file, (err, stats) ->
+        return done(err) if err || !stats.isFile()
+        exports.uploadAsset({cwd, design, host, token, file}, done)
+    , callback
+
+
+exports.uploadAsset = ({cwd, design, host, token, file}, callback) ->
+  relativePath = file.replace(cwd, '')
+  request
+    method: 'post'
+    url: host+"/designs/#{design.name}/#{design.version}/assets"
+    headers:
+      Authorization: "Bearer #{token}"
+    formData:
+      path: relativePath
+      file: fs.createReadStream(file)
+  , (err, res, body) ->
+    return callback(err) if err
+    if res.statusCode == 200
+      console.log("Uploaded the file '#{relativePath}'")
```

Comment:
Migrating that file to use an EventEmitter would be good idea as it allows to send events instead of logging something to the console.